### ISSUE TITLE
https://checkstyle.org/checks/ Gives a 404

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#13494]
https://checkstyle.org/checks/
````
File not found

The site configured at this address does not contain the requested file.

If this is your site, make sure that the filename case matches the URL as well as any file permissions.
````

We are missing an index file.

There is no link to this page that I am aware. I just noticed there was no index.

---------

All our issues have links to Checks and all of them are invalid.
I think we need to set redirects 
Of https://checkstyle.org/config_javadoc.html#JavadocType
To
https://checkstyle.org/checks/javadoc/index.html or better to https://checkstyle.org/checks/javadoc/javadoctype.html

Redirects in GitHub io: https://dev.to/steveblue/setup-a-redirect-on-github-pages-1ok7 is simply html page with redirect link.
https://gist.github.com/domenic/1f286d415559b56d725bee51a62c24a7

Might be good - Redirects of all non existing pages to some predefined https://stackoverflow.com/a/66716264/1015848

------
we need to fix all links to Checks from old url to new url
if you look at any issue before 2023, like https://github.com/checkstyle/checkstyle/issues/13155 (see https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance)
you can see link to Check , but it is not working anymore
as we changed layout of website
whole our issue tracker and all links is whole web become invalid
for exmaple in Stackoverflow.com
task is put page https://checkstyle.org/config_coding.html
that is taking archor from url #VariableDeclarationUsageDistance
that auto redirect https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance to https://checkstyle.org/checks/coding/variabledeclarationusagedistance.html#VariableDeclarationUsageDistance
so all links in web (forums, ....) become good again

**Comment by rnveach** [comment]
I recommend just to redirect to https://checkstyle.org/checks.html .

**Comment by romani** [comment]
We will see how to do this

Some redirection:
https://github.com/checkstyle/checkstyle/blob/2bd1a8bb63bcf0f4e5e90fb52e4c5aaccf53657b/.ci/release-copy-github-io-to-sourceforge.sh#L62

**Comment by romani** [comment]
All our issues have links to Checks and all of them are invalid.
I think we need to set redirects 
Of https://checkstyle.org/config_javadoc.html#JavadocType
To
https://checkstyle.org/checks/javadoc/index.html

**Comment by harshit4311** [comment]
I would love working on this issue! Could you please assign it to me?

**Comment by romani** [comment]
No assignments, just send PR.

**Comment by MANISH-K-07** [comment]
@romani , @rnveach , I've waited for a month but there hasn't been any activity from the author. I would like to fix this issue.

I'm not that familiar with gH pages. I believe, we must set up an entry file (like README , or index in our case), am I right ?
This index is a common html file that redirects the gH page to custom set Url.
I think this index.html should be in the top-level for the dir. So, for https://checkstyle.org/checks/ , we need to add an index file in our repo at https://github.com/checkstyle/checkstyle/tree/master/src/xdocs/checks ??

Please point me to correct path where we need index, in case my understanding is wrong

**Comment by romani** [comment]
> we need to add an index file in our repo at https://github.com/checkstyle/checkstyle/tree/master/src/xdocs/checks ??

yes.
and this index file should redirect to https://checkstyle.org/checks.html

The same problem we have with old links to Checks, example https://github.com/checkstyle/checkstyle/issues/13155, usage  of such links in web in enourous, and even in our issue tracker, so we did bad job to damage them, all our old issues are damaged.
we need have old pages redirect 
https://checkstyle.sourceforge.io/config_coding.html#VariableDeclarationUsageDistance
to
https://checkstyle.org/checks/coding/variabledeclarationusagedistance.html#VariableDeclarationUsageDistance

Solution with recreation of pages `/config_coding.html` with empty content and JS to do proper redirect works for me.

**Comment by MANISH-K-07** [comment]
> yes.
> and this index file should redirect to https://checkstyle.org/checks.html

Yep thanks, I understood that part. 
So, I will first send a PR for the parent dir (src/checks) such that  https://checkstyle.org/checks/ redirects to https://checkstyle.org/checks.html
If all is good, We will fix rest of the broken page links ;)

@romani , One minor thing, could you please explain how https://github.com/checkstyle/checkstyle/issues/13494#issuecomment-1659529907 is related here ?

**Comment by romani** [comment]
this will works only for sourceforge hosting, where we can specify `.htaccess` file.
It will not work for github pages hosting, so Javascript probably is only a solution.

https://stackoverflow.com/questions/13446435/is-there-an-alternative-to-htaccess-for-github-pages
https://docs.oracle.com/cd/B14099_19/web.1012/q20206/howto/htaccess.html

**Comment by romani** [comment]
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/ded61f8_2024030159/checks/ not working

**Comment by MANISH-K-07** [comment]
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/ded61f8_2024030159/checks/ not working

Looks like the client is parsing file content as xml instead of html. I will try to debug the http response and get back

**Comment by Zopsss** [comment]
I'm trying to work on this issue, following up @MANISH-K-07's PR: https://github.com/checkstyle/checkstyle/pull/14667

**Comment by nrmancuso** [comment]
Closed via #15949

**Comment by romani** [comment]
@Zopsss, thanks a lot for your fix.
I verified after release, old links works!!
